string.c

```diff
@@ -7695,6 +7695,9 @@ rb_str_crypt(VALUE str, VALUE salt)
     }
 
     s = RSTRING_PTR(str);
```

Comment:
As @matthewd said, this line should be changed to `s = StringValueCStr(str);` instead of your lines below.